harness/determined/pytorch/lightning/_adapter.py

```diff
@@ -152,16 +171,37 @@ def __init__(self, context: PyTorchTrialContext) -> None:
         optimizers, _ = self.setup_optimizers_schedulers()
         pls.optimizers = optimizers
 
+        # CHECK our `check` usage vs raising exceptions seems inconsistent.
+        check.check_in(precision, {16, 32}, "only precisions 16 & 32 are supported.")
+
+        use_amp = context.experimental._auto_amp or context._use_apex
```

Comment:
This seems wrong. Because `_use_apex` is set in `configure_apex_amp` and the same for `_auto_amp`

Reply:
I'm trying to set use_amp without relying on the precision property 🤔 let me see